### PR TITLE
Moving FocusTrapZone and Modal from react-next to react-internal

### LINKED ISSUE #15266: Port aria-hidden siblings change to next
#13281 is missing from react-next across multiple components. The prop is present but unused in Modal, and entirely missing from FocusTrapZone.

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [x] Addresses an existing issue: Part of #15152, fixes #15266
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This PR moves the versions of `Modal` and `FocusTrapZone` existing in `@fluentui/react-next` to `@fluentui/react-internal`. As part of this API and snapshots were updated, the `@fluentui/react-next` examples, vr tests and perf tests of these components were removed, and the versions of these components in `@fluentui/react-next` now re-export the versions in `@fluentui/react-internal`.

Another change in this PR is that `resetIds` is now being called in those tests that were affected by this change.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/react-internal/src/components/Dialog/__snapshots__/Dialog.deprecated.test.tsx.snap [hunk 1]
```diff
@@ -371,8 +369,8 @@ exports[`Dialog deprecated props renders Dialog with containerClassName 1`] = `
     onTouchStart={[Function]}
   >
     <div
-      aria-describedby="Dialog2-subText"
-      aria-labelledby="Dialog2-title"
+      aria-describedby="Dialog3-subText"
```
Comment (ecraig12345) [inline packages/react-internal/src/components/Dialog/__snapshots__/Dialog.deprecated.test.tsx.snap:372]: Looks like Dialog isn't calling resetIds--if you don't mind, would be good to fix that (in both react-internal and react-next so it doesn't get lost). Should probably be done in beforeEach and afterAll.

**Comment by ecraig12345** [comment]
Can you delete the FocusTrapZoneNext perf-tests and vr-tests?

File: packages/react-internal/src/components/Modal/Modal.base.tsx [hunk 2]
```diff
@@ -464,25 +413,25 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
         }
         case KeyCodes.up: {
           this.setState({
-            y: this._getClampedPositionY(this.state.y - delta),
+            y: this.state.y - delta,
```
Comment (ecraig12345) [inline packages/react-internal/src/components/Modal/Modal.base.tsx:416]: The clamping is a new change which hasn't been ported. We should definitely not remove that, and if it involved any new props to FocusTrapZone we should add those too.

File: packages/react-internal/src/components/Modal/Modal.base.tsx [hunk 3]
```diff
@@ -470,19 +467,19 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
...
           this.setState({
-            x: this._getClampedPositionX(this.state.x - delta),
+            x: this._getClampedPositionY(this.state.x - delta),
           });
           break;
         }
         case KeyCodes.right: {
           this.setState({
-            x: this._getClampedPositionX(this.state.x + delta),
+            x: this._getClampedPositionY(this.state.x - delta),
```
Comment (ecraig12345) [inline packages/react-internal/src/components/Modal/Modal.base.tsx:482]: Several typos here

File: packages/react-internal/src/components/Modal/Modal.base.tsx [hunk 4]
```diff
@@ -152,9 +154,6 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
         isVisible: true,
       });
     }
-    if (!prevProps.isOpen && this.props.isOpen) {
```
Comment (ecraig12345) [inline packages/react-internal/src/components/Modal/Modal.base.tsx:155]: I'm pretty sure that all the changes to this file should be reverted

File: packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.types.ts [hunk 5]
```diff
@@ -78,11 +78,4 @@ export interface IFocusTrapZoneProps extends React.HTMLAttributes<HTMLDivElement
    * @defaultvalue false
    */
   focusPreviouslyFocusedInnerElement?: boolean;
-
-  /**
-   * Puts aria-hidden=true on all non-ancestors of the current element, for screen readers.
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.types.ts:83]: This is new and needs to come back.

**Review by ecraig12345 — COMMENTED** [review]
Thanks for doing this, but it looks like we're

File: packages/react-internal/src/components/Dialog/__snapshots__/Dialog.deprecated.test.tsx.snap [hunk 6]
```diff
@@ -91,7 +91,6 @@ exports[`Dialog deprecated props renders Dialog with className 1`] = `
               transition: opacity 0.267s;
               width: 100%;
             }
-        role="document"
```
Comment (ecraig12345) [inline packages/react-internal/src/components/Dialog/__snapshots__/Dialog.deprecated.test.tsx.snap:94]: Might be missing a snapshot update? I wouldn't expect this change now that the Modal changes have been reverted.

Comment (khmakoto) [reply packages/react-internal/src/components/Dialog/__snapshots__/Dialog.deprecated.test.tsx.snap]: Updated!

**Review by tomi-msft — APPROVED** [review]

File: packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.tsx [hunk 7]
```diff
@@ -309,9 +313,15 @@ export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
       if (!disableFirstFocus && !elementContains(root.current, internalState.previouslyFocusedElementOutsideTrapZone)) {
         focus();
       }
+      if (!unmodalize && root.current && enableAriaHiddenSiblings) {
+        unmodalize = modalize(root.current);
+      }
     } else if (!newForceFocusInsideTrap || newDisabled) {
       // Transition from forceFocusInsideTrap / FTZ enabled to disabled.
       returnFocusToInitiator();
+      if (unmodalize) {
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.tsx:322]: Thanks for adding this part! However I think it needs to be handled a bit differently to replicate the class component behavior: the `unmodalize` callback should be stored in `internalState`, and in addition to this part you'll need to add another useEffect for componentWillUnmount.

I started trying to make this change last night and then noticed potentially a bunch of other issues with lifecycles that may not have been ported properly...so we definitely want to test this one carefully.

Comment (khmakoto) [reply packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.tsx:322]: I think I did this properly but let me know which other issues you've found.

**Review by ecraig12345 — APPROVED** [review]

**Review by xugao — APPROVED** [review]

**Pull request merged**